src/cosmic/bplane.rs

```diff
@@ -318,108 +318,81 @@ pub fn try_achieve_b_plane(
-
-            if br_err.abs() < target.tol_b_r_km && bt_err.abs() < target.tol_b_t_km {
-                return Ok((total_dv, b_plane));
-            }
-
-            // Build the error vector
-            let b_plane_err = Vector2::new(br_err, bt_err);
-
-            if b_plane_err.norm() >= prev_b_plane_err {
-                // If the error is not going down, we'll raise an error
-                return Err(TargetingError::CorrectionIneffective {
-                    prev_val: prev_b_plane_err,
-                    cur_val: b_plane_err.norm(),
-                    action: "Delta-V correction is ineffective at reducing the B-Plane error",
-                });
-            }
-            prev_b_plane_err = b_plane_err.norm();
-
-            // Grab the first two rows of the Jacobian (discard the rest).
-            let full_jac = b_plane.jacobian();
-            let jac = full_jac.fixed_rows::<2>(0);
-            // Solve the Least Squares / compute the delta-v
-            let dv = jac.transpose() * (jac * jac.transpose()).try_inverse().unwrap() * b_plane_err;
-
-            total_dv[0] += dv[0];
-            total_dv[1] += dv[1];
-            total_dv[2] += dv[2];
-
-            // Rebuild a new orbit
-            real_orbit.vx_km_s += dv[0];
-            real_orbit.vy_km_s += dv[1];
-            real_orbit.vz_km_s += dv[2];
-
-            attempt_no += 1;
+    for _ in 0..=MAX_ITER {
+        // Build current B Plane
+        let b_plane = BPlane::new(real_orbit).with_context(|_| AstroSnafu)?;
+
+        // Check convergence
+        let br_err = target.b_r_km - b_plane.b_dot_r();
+        let bt_err = target.b_t_km - b_plane.b_dot_t();
+        let ltof_err = target
+            .ltof_target_set()
+            .then_some(target.ltof_s - b_plane.ltof_s.real());
+
+        if br_err.abs() < target.tol_b_r_km
+            && bt_err.abs() < target.tol_b_t_km
+            && !ltof_err.is_some_and(|err| err.abs() >= target.tol_ltof_s)
+        {
+            return Ok((total_dv, b_plane));
         }
-    } else {
-        // The LTOF targeting seems to break often, but it's still implemented
-        loop {
-            if attempt_no > max_iter {
-                return Err(TargetingError::TooManyIterations);
-            }
 
-            // Build current B Plane
-            let b_plane = BPlane::new(real_orbit).with_context(|_| AstroSnafu)?;
-
-            // Check convergence
-            let br_err = target.b_r_km - b_plane.b_dot_r();
-            let bt_err = target.b_t_km - b_plane.b_dot_t();
-            let ltof_err = target.ltof_s - b_plane.ltof_s.real();
-
-            if br_err.abs() < target.tol_b_r_km
-                && bt_err.abs() < target.tol_b_t_km
-                && ltof_err.abs() < target.tol_ltof_s
-            {
-                return Ok((total_dv, b_plane));
+        let (norm, dv) = match ltof_err {
+            None => {
+                let v = Vector2::new(br_err, bt_err);
```

Comment:
Oh, good find! The B-Plane Jacobian is indeed BR, BT, LTOF. That said, I think this should be fixed in the Jacobian() function so it returns BT, BR, LTOF as does the Jacobian2 function and as the documentation says.

https://rustdoc.nyxspace.com/src/nyx_space/cosmic/bplane.rs.html#140-152